doc/classes/Array.xml

```diff
@@ -142,7 +142,25 @@
 			<argument index="3" name="before" type="bool" default="true">
 			</argument>
 			<description>
-				Finds the index of an existing value (or the insertion index that maintains sorting order, if the value is not yet present in the array) using binary search and a custom comparison method. Optionally, a [code]before[/code] specifier can be passed. If [code]false[/code], the returned index comes after all existing entries of the value in the array. The custom method receives two arguments (an element from the array and the value searched for) and must return [code]true[/code] if the first argument is less than the second, and return [code]false[/code] otherwise.
+				Finds the index of an existing value (or the insertion index that maintains sorting order, if the value is not yet present in the array) using binary search and a custom comparison method declared in the [code]obj[/code]. Optionally, a [code]before[/code] specifier can be passed. If [code]false[/code], the returned index comes after all existing entries of the value in the array. The custom method receives two arguments (an element from the array and the value searched for) and must return [code]true[/code] if the first argument is less than the second, and return [code]false[/code] otherwise.
+				[codeblock]
+				extends Node
+				func cardinal_to_algebraic(a):
+				    match a:
+				        "one":
+				            return 1
+				        "two":
+				            return 2
+				        "three":
+				            return 3
+				        _:
+				            return 4
+				func compare(a,b): ## This compare is called when obj is set to self
+				    return cardinal_to_algebraic(a) &lt; cardinal_to_algebraic(b)
```

Comment:
@akien-mga , Regardless of object type, bsearch_custom is a flexible arbitrary function. I believe creating these odd examples help beginners understand they can use these functions beyond the normal expectation.

Reply:
Yeah that's a good example now I think.

Reply:
I changed it to different languages. I will squash my commits

Reply:
That made it more confusing IMO. And I don't like using `_` to match actual valid values that just happen to be `3`. `_` should be used to match invalid input, and should likely give 0 or -1, as I suggested in my previous edit.

And please apply the style changes that I suggested twice already.